### DISCUSSION TITLE
Get rotors RPM data from Crazyflie 2.1 on Crazyswarm and visualizing control methods

### DISCUSSION THREAD

**Discussion by yusufbolat** [USC-ACTLab/crazyswarm#418] [Q&A] [answered]
Hi There,

How can we get rotors RPM data from crazyflies? Also we would like to know how to visualize Crazyswarm control methods in Gazebo (using CrazyS) ?

**Comment by ataffanel** [comment, marked as answer]
I do not know much about Gazebo but for the RPM data, there is no way to get them from the Crazyflie since the Crazyflie does not measure the motor RPM. You can get the PWM applied to the motors by logging the variables ["motor.m[1-4]"](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#motor). The RPM and thrust can be estimated from [PWM and battery voltage](https://www.bitcraze.io/documentation/wiki/pwm-to-thrust/).

**Reply by whoenig** [reply]
There is no integration of CrazyS and Crazyswarm at the moment.

**Reply by yusufbolat** [reply]
Thanks for reply. How can we apply PWM data on CrazyS models? (for virtual-reality integration)

**Reply by whoenig** [reply]
You would probably need to ask the maintainers of the CrazyS project.
